pkg/services/authn/clients/proxy_test.go

```diff
@@ -177,14 +178,120 @@ func TestProxy_Test(t *testing.T) {
 var _ proxyCache = new(fakeCache)
 
 type fakeCache struct {
-	expectedErr  error
-	expectedItem []byte
+	data        map[string][]byte
+	expectedErr error
 }
 
-func (f fakeCache) Get(ctx context.Context, key string) ([]byte, error) {
-	return f.expectedItem, f.expectedErr
+func (f *fakeCache) Get(ctx context.Context, key string) ([]byte, error) {
+	return f.data[key], f.expectedErr
 }
 
-func (f fakeCache) Set(ctx context.Context, key string, value []byte, expire time.Duration) error {
+func (f *fakeCache) Set(ctx context.Context, key string, value []byte, expire time.Duration) error {
+	f.data[key] = value
 	return f.expectedErr
 }
+
+func (f fakeCache) Delete(ctx context.Context, key string) error {
+	delete(f.data, key)
+	return f.expectedErr
+}
+
+func TestProxy_Hook(t *testing.T) {
```

Comment:
Great, one note though. I would prefer not to use a table test here. They are used for different test cases but here is just one test with different input, it would also save a a lot of boiler plate code this now need to have

Reply:
Sorry, I don't know about the table test, could you show me a example, like another unit test or link to an article maybe?

Reply:
No need to apologize. What has the structure of a table test but usually the fist call in the loop would be something like 
```go
for _, tt := range tests {
  t.Run(tt.desc, func (t *testing.T) {
     // test code here
  })
}
``` 
Then they would be executed as separate tests. These test has that structure but is actually just one test to make sure we won't have a regression on this. 

So instead of having testCase struct here and the loop just write it as a normal test, something like

```go
func TestProxy_hook(t *testing-T) {
  // initilize cfg, cache, proxy client here so we can share the state
  
  t.Run("step 1", func (t *testing.T) {

  })
  
  t.Run("step 2", func (t *testing.T) {

  })
  
  t.Run("step 3", func (t *testing.T) {

  })
}
```
But use a more descriptive name passed to each test function

Reply:
Aah, I see. Thanks for the explanation.